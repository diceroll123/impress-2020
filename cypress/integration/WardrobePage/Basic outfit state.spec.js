// Give network requests a bit of breathing room!
const networkTimeout = { timeout: 12000 };

describe("WardrobePage: Basic outfit state", () => {
  it("Initialize simple outfit from URL", () => {
    cy.visit("/outfits/new?species=1&color=8&objects[]=76789");

    getSpeciesSelect(networkTimeout)
      .find(":selected")
      .should("have.text", "Acara");
    getColorSelect().find(":selected").should("have.text", "Blue");
    cy.location().toMatchSnapshot();

    cy.contains("A Warm Winters Night Background", networkTimeout).should(
      "exist"
    );

    getOutfitPreview().toMatchImageSnapshot();
  });

  it("Changes species and color", () => {
    cy.visit("/outfits/new?species=1&color=8&objects[]=76789");

    getSpeciesSelect(networkTimeout)
      .find(":selected")
      .should("have.text", "Acara");
    getColorSelect().find(":selected").should("have.text", "Blue");
    cy.location().toMatchSnapshot();
    getOutfitPreview().toMatchImageSnapshot();

    getSpeciesSelect().select("Aisha");

    getSpeciesSelect().find(":selected").should("have.text", "Aisha");
    getColorSelect().find(":selected").should("have.text", "Blue");
    cy.location().toMatchSnapshot();
    getOutfitPreview().toMatchImageSnapshot();

    getColorSelect().select("Red");

    getSpeciesSelect().find(":selected").should("have.text", "Aisha");
    getColorSelect().find(":selected").should("have.text", "Red");
    cy.location().toMatchSnapshot();
    getOutfitPreview().toMatchImageSnapshot();
  });

  it.only("Changes pose", () => {
    cy.visit("/outfits/new?species=1&color=8&pose=HAPPY_FEM");

    getPosePickerButton(networkTimeout).click();
    getPosePickerOption("Happy and Feminine").should("be.checked");
    cy.location().toMatchSnapshot();
    getOutfitPreview().toMatchImageSnapshot();

    getPosePickerOption("Sad and Masculine").check({ force: true });
    getPosePickerOption("Sad and Masculine").should("be.checked");
    cy.location().toMatchSnapshot();
    getOutfitPreview().toMatchImageSnapshot();
  });

  it("Toggles item", () => {
    cy.visit("/outfits/new?species=1&color=8&objects[]=76789");

    getOutfitPreview().toMatchImageSnapshot();
    cy.location().toMatchSnapshot();

    cy.contains("A Warm Winters Night Background").click();

    getOutfitPreview().toMatchImageSnapshot();
    cy.location().toMatchSnapshot();
  });

  it("Renames outfit", () => {
    cy.visit("/outfits/new?name=My+outfit&species=1&color=8");

    getOutfitName(networkTimeout).should("have.text", "My outfit");

    getOutfitName().click().type("Awesome outfit{enter}");

    getOutfitName().should("have.text", "Awesome outfit");
    cy.location().toMatchSnapshot();
  });
});

function getSpeciesSelect(options) {
  return cy.get("[data-test-id=wardrobe-species-picker]", options);
}

function getColorSelect(options) {
  return cy.get("[data-test-id=wardrobe-color-picker]", options);
}

function getPosePickerButton(options) {
  return cy.get("[data-test-id=wardrobe-pose-picker]", options);
}

function getPosePickerOption(label, options) {
  return cy.get(`input[aria-label="${CSS.escape(label)}"]`, options);
}

function getOutfitPreview() {
  return cy.get("[data-test-id=wardrobe-outfit-preview]:not([data-loading])", {
    // A bit of an extra-long timeout, to await both server data and image data
    timeout: 15000,
  });
}

function getOutfitName(options) {
  return cy.get("[data-test-id=outfit-name]", options);
}
