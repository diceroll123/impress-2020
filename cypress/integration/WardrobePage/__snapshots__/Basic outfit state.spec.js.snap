exports[`WardrobePage: Basic outfit state > Initialize simple outfit from URL #0`] =
{
  "auth": "",
  "hash": "",
  "host": "localhost:3000",
  "hostname": "localhost",
  "href": "http://localhost:3000/outfits/new?name=&species=1&color=8&pose=HAPPY_FEM&objects%5B%5D=76789",
  "origin": "http://localhost:3000",
  "originPolicy": "http://localhost:3000",
  "pathname": "/outfits/new",
  "port": "3000",
  "protocol": "http:",
  "search": "?name=&species=1&color=8&pose=HAPPY_FEM&objects%5B%5D=76789",
  "superDomain": "localhost"
};

exports[`WardrobePage: Basic outfit state > Changes species and color #0`] =
{
  "auth": "",
  "hash": "",
  "host": "localhost:3000",
  "hostname": "localhost",
  "href": "http://localhost:3000/outfits/new?name=&species=1&color=8&pose=HAPPY_FEM&objects%5B%5D=76789",
  "origin": "http://localhost:3000",
  "originPolicy": "http://localhost:3000",
  "pathname": "/outfits/new",
  "port": "3000",
  "protocol": "http:",
  "search": "?name=&species=1&color=8&pose=HAPPY_FEM&objects%5B%5D=76789",
  "superDomain": "localhost"
};

exports[`WardrobePage: Basic outfit state > Changes species and color #1`] =
{
  "auth": "",
  "hash": "",
  "host": "localhost:3000",
  "hostname": "localhost",
  "href": "http://localhost:3000/outfits/new?name=&species=2&color=8&pose=HAPPY_FEM&objects%5B%5D=76789",
  "origin": "http://localhost:3000",
  "originPolicy": "http://localhost:3000",
  "pathname": "/outfits/new",
  "port": "3000",
  "protocol": "http:",
  "search": "?name=&species=2&color=8&pose=HAPPY_FEM&objects%5B%5D=76789",
  "superDomain": "localhost"
};

exports[`WardrobePage: Basic outfit state > Changes species and color #2`] =
{
  "auth": "",
  "hash": "",
  "host": "localhost:3000",
  "hostname": "localhost",
  "href": "http://localhost:3000/outfits/new?name=&species=2&color=61&pose=HAPPY_FEM&objects%5B%5D=76789",
  "origin": "http://localhost:3000",
  "originPolicy": "http://localhost:3000",
  "pathname": "/outfits/new",
  "port": "3000",
  "protocol": "http:",
  "search": "?name=&species=2&color=61&pose=HAPPY_FEM&objects%5B%5D=76789",
  "superDomain": "localhost"
};

exports[`WardrobePage: Basic outfit state > Toggles item #0`] =
{
  "auth": "",
  "hash": "",
  "host": "localhost:3000",
  "hostname": "localhost",
  "href": "http://localhost:3000/outfits/new?name=&species=1&color=8&pose=HAPPY_FEM&objects%5B%5D=76789",
  "origin": "http://localhost:3000",
  "originPolicy": "http://localhost:3000",
  "pathname": "/outfits/new",
  "port": "3000",
  "protocol": "http:",
  "search": "?name=&species=1&color=8&pose=HAPPY_FEM&objects%5B%5D=76789",
  "superDomain": "localhost"
};

exports[`WardrobePage: Basic outfit state > Toggles item #1`] =
{
  "auth": "",
  "hash": "",
  "host": "localhost:3000",
  "hostname": "localhost",
  "href": "http://localhost:3000/outfits/new?name=&species=1&color=8&pose=HAPPY_FEM&closet%5B%5D=76789",
  "origin": "http://localhost:3000",
  "originPolicy": "http://localhost:3000",
  "pathname": "/outfits/new",
  "port": "3000",
  "protocol": "http:",
  "search": "?name=&species=1&color=8&pose=HAPPY_FEM&closet%5B%5D=76789",
  "superDomain": "localhost"
};

exports[`WardrobePage: Basic outfit state > Changes pose #0`] =
{
  "auth": "",
  "hash": "",
  "host": "localhost:3000",
  "hostname": "localhost",
  "href": "http://localhost:3000/outfits/new?name=&species=1&color=8&pose=HAPPY_FEM",
  "origin": "http://localhost:3000",
  "originPolicy": "http://localhost:3000",
  "pathname": "/outfits/new",
  "port": "3000",
  "protocol": "http:",
  "search": "?name=&species=1&color=8&pose=HAPPY_FEM",
  "superDomain": "localhost"
};

exports[`WardrobePage: Basic outfit state > Changes pose #1`] =
{
  "auth": "",
  "hash": "",
  "host": "localhost:3000",
  "hostname": "localhost",
  "href": "http://localhost:3000/outfits/new?name=&species=1&color=8&pose=SAD_MASC",
  "origin": "http://localhost:3000",
  "originPolicy": "http://localhost:3000",
  "pathname": "/outfits/new",
  "port": "3000",
  "protocol": "http:",
  "search": "?name=&species=1&color=8&pose=SAD_MASC",
  "superDomain": "localhost"
};
